src/Microsoft.AspNetCore.Mvc.TagHelpers/DependencyInjection/TagHelperExtensions.cs

```diff
@@ -32,7 +35,10 @@ public static IMvcCoreBuilder AddCacheTagHelper(this IMvcCoreBuilder builder)
 
             // Required default services for cache tag helpers
             builder.Services.TryAddSingleton<IDistributedCache, MemoryDistributedCache>();
-            builder.Services.TryAddSingleton<IMemoryCache, MemoryCache>();
+            builder.Services.TryAddSingleton<CacheTagHelperMemoryCacheFactory>();
+            //builder.Services.Configure<MemoryCacheOptions>(
+            //    CacheTagHelperMemoryCacheFactory.MemoryCacheOptionsName,
+            //    o => o.EntryCountLimit = DefaultEntryCountLimit);
```

Comment:
Let's go ahead with this PR without waiting for `EntryCountLimit` to be added. I want to get as much of this in today as possible.